core/src/main/java/org/ehcache/config/ResourcePoolsBuilder.java

```diff
@@ -38,6 +38,15 @@ public static ResourcePoolsBuilder newResourcePoolsBuilder() {
     return new ResourcePoolsBuilder();
   }
 
+  public static ResourcePoolsBuilder newResourcePoolsBuilder(ResourcePools pools) {
+    ResourcePoolsBuilder poolsBuilder = new ResourcePoolsBuilder();
+    for (ResourceType currentResourceType : pools.getResourceTypeSet()) {
+      poolsBuilder = poolsBuilder.with(currentResourceType, pools.getPoolForResource(currentResourceType).getSize(),
+          pools.getPoolForResource(currentResourceType).getUnit(), currentResourceType.isPersistable());
```

Comment:
It's not because the resource type is persistable that it has been declared persistent. Read the value from the `ResourcePool` - which you should extract to a local var to not have to call it three times.